airflow-core/src/airflow/api_fastapi/core_api/datamodels/ui/common.py

```diff
@@ -77,7 +77,7 @@ class GridRunsResponse(BaseModel):
     start_date: datetime | None
     end_date: datetime | None
     run_after: datetime
-    state: TaskInstanceState | None
+    state: DagRunState | None
```

Comment:
Uuups, good catch!

Important fix but... @bbovenzi /  @pierrejeambrun  this change is a bit of a breaking change. Must be prioritize this change for 3.1 release?

Reply:
@kaxil FYI might be we need to cherry-pick this line as "interface bug fix) (without the other feature but no harm to add it late) to 3.1.0 as the wrong type is on the public API